br/pkg/lightning/backend/kv/base.go

```diff
@@ -93,6 +94,13 @@ func (row RowArrayMarshaller) MarshalLogArray(encoder zapcore.ArrayEncoder) erro
 			if err != nil {
 				return err
 			}
+			if len(str) > 512*1024 {
+				str = str[0:1024]
```

Comment:
maybe str = str[0:1024]+"<truncated>" so for future developer will know it's truncated rather than the datum is corrupted